src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -43,11 +49,61 @@ define([
         deploymentsFetchState: States.STATE_LOADING,
         fetchState: States.STATE_LOADING,
         modalClass: null,
+        route: null,
         tasksFetchState: States.STATE_LOADING
       };
     },
 
     componentDidMount: function() {
+      var router = this.props.router;
+
+      function callRoute(routeFunc) {
+        if(this.state.fetchState === States.STATE_LOADING) {
+          setTimeout(callRoute.bind(this, routeFunc), 300);
+        } else {
+          routeFunc();
+        }
+      };
+
+      var destroyModal = function() {
+        if(this.state.modalClass) {
+          this.modalDestroy();
+        }
+      }.bind(this);
+
+      // set route in state for all pages
+      router.on("route", function (route, params) {
+        var routeName = _.str.capitalize(route);
+
+        // Modals
+        if(route === "about" || route === "app" || route === "newapp") {
+          if(route !== router.lastRoute.route || params[0] !== router.lastRoute.params[0]) {
+            destroyModal();
+          }
+          callRoute.call(this, this["route" + routeName].bind(this, params));
```

Comment:
This function should be called in render and it should only result in a component. You only wanna change state in `componentDidMount`